src/modules/booking/booking.service.ts

```diff
@@ -110,4 +108,107 @@ export class BookingService {
       throw new BadRequestException('Booking dates cannot be in the past');
     }
   }
+
+  async getBookingById(id: number): Promise<GetBookingResponse> {
+    const booking = await this.prismaService.booking.findUnique({
+      where: { id },
+    });
+
+    if (!booking) {
+      throw new NotFoundException('Booking not found');
+    }
+
+    return booking;
+  }
+
+  async getBookingsByUserId(userId: number): Promise<GetBookingResponse[]> {
+    const bookings = await this.prismaService.booking.findMany({
+      where: { userId },
+    });
+
+    if (!bookings || bookings.length === 0) {
+      throw new NotFoundException('No bookings found for the user');
+    }
+
+    return bookings;
+  }
+
+  async updateBooking(
+    id: number,
+    userId: number,
+    newBookingData: UpdateBookingDto,
+  ): Promise<CreateBookingResponse> {
+    const booking = await this.prismaService.booking.findUnique({
+      where: { id },
+      include: { room: true, user: true },
+    });
+
+    if (!booking) {
+      throw new NotFoundException('Booking not found');
+    }
+
+    if (booking.userId !== userId) {
+      throw new BadRequestException(
+        'You are not authorized to update this booking',
+      );
+    }
+
+    const { roomTypeId, checkIn, checkOut } = newBookingData;
```

Comment:
I would not do this, it is more understandable if you use newBookingData.checkIn, newBookingData.checkOut and so on